### PR TITLE
HTCONDOR-981 Implement `htcondor annex create`

### PR DESCRIPTION
**Pull request by jasoncpatton**
https://opensciencegrid.atlassian.net/browse/HTCONDOR-981

Docs and tests to come later in HTCONDOR-958

# HTCondor Pull Request Checklist for internal reviewers

- [x] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [x] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [x] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [x] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [x] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [x] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [x] Check for documentation, if needed
- [x] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [x] Check that commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

**Comment by Todd-L-Miller** [comment]
Documention, version history, and regression tests will be done in HTCONDOR-958 as appropriate.

File: src/condor_tools/htcondor_cli/annex_create.py [hunk 1]
```diff
@@ -0,0 +1,720 @@
...
+
+INITIAL_CONNECTION_TIMEOUT = int(
+    htcondor.param.get("ANNEX_INITIAL_CONNECTION_TIMEOUT", 180)
+)
+REMOTE_CLEANUP_TIMEOUT = int(htcondor.param.get("ANNEX_REMOTE_CLEANUP_TIMEOUT", 60))
+REMOTE_MKDIR_TIMEOUT = int(htcondor.param.get("ANNEX_REMOTE_MKDIR_TIMEOUT", 5))
+REMOTE_POPULATE_TIMEOUT = int(htcondor.param.get("ANNEX_REMOTE_POPULATE_TIMEOUT", 60))
+
+# FIXME: Make this come from config? At least some place not hardcoded.
+MACHINE_QUEUE_MAP = {"stampede2": "development"}
```
Comment (Todd-L-Miller) [inline src/condor_tools/htcondor_cli/annex_create.py:26]:
```suggestion
MACHINE_QUEUE_MAP = {"stampede2": "normal"}
```
The `development` queue turns around faster but only allows one job.

File: src/condor_tools/htcondor_cli/annex_create.py [hunk 2]
```diff
@@ -0,0 +1,720 @@
...
+        annex_name,
+        queue_name,
+        collector,
+        str(remote_script_dir / token_file.name),
+        str(lifetime),
+        str(remote_script_dir / f"{target}.pilot"),
+        owners,
+        str(nodes),
+        str(remote_script_dir / f"{target}.multi-pilot"),
+        str(allocation),
```
Comment (Todd-L-Miller) [inline src/condor_tools/htcondor_cli/annex_create.py:282]: Didn't we decide that this had to be `f'{allocation}'` to behave properly if `allocation` were `None`?

**Review by Todd-L-Miller — COMMENTED** [review]
* I like sticking the HPC annex -specific files in a subdirectory of `libexec/`.
* The files in `annex/` are direct imports from https://github.com/CHTC/hpc-annex, which is proper, and I like that they're in the own subdirectory in the source, too.  Since i wrote them, I won't review them further. ;)
* `annex_create.py` is a revised-and-improved version of `hpc-annex.py` from the aformentioned repository.

I think you missed one fix here; see comment below.  Otherwise, this looks good.

File: src/condor_tools/htcondor_cli/annex_create.py (hunk 2 above)
Comment (Todd-L-Miller) [reply src/condor_tools/htcondor_cli/annex_create.py:282]: (To whit, `stampede.sh` depends on `allocation` showing up as the string `None` on the command-line to preserve parameter indices.)

**Comment by Todd-L-Miller** [comment]
Also, although I didn't notice until just now, you forgot to wire the new verb into the `annex` noun.

File: src/condor_tools/htcondor_cli/annex_create.py (hunk 2 above)
Comment (jasoncpatton) [reply src/condor_tools/htcondor_cli/annex_create.py:282]: It's the same:
```python
>>> allocation = None
>>> str(allocation) == f"{allocation}"
True
>>> str(allocation)
'None'
```

Comment (Todd-L-Miller) [reply src/condor_tools/htcondor_cli/annex_create.py:282]: Sorry.  You'd actually told me that before and I forgot.

**Review by Todd-L-Miller — COMMENTED** [review]
Looks good.

**Review by Todd-L-Miller — APPROVED** [review]
This tests cleanly for me now.

**Pull request merged**